plugin.video.southpark_unofficial/southpark.py

```diff
@@ -0,0 +1,596 @@
+			self.notify(self.i18n.WARNING_GEOBLOCKED, WARNING_TIMEOUT_LONG)
+		
+		self.add_directory(self.i18n.MENU_FEATURED_EPISODES, '', PLUGIN_MODE_FEATURED, self.paths.PLUGIN_ICON)
+		self.add_entry    (self.i18n.MENU_RANDOM_EPISODE   , '', PLUGIN_MODE_RANDOM  , self.paths.PLUGIN_ICON, is_playable=self.options.playrandom())
+		self.add_entry    (self.i18n.MENU_SEARCH_EPISODE   , '', PLUGIN_MODE_SEARCH  , self.paths.PLUGIN_ICON)
+		for i in range(1, self.seasons):
+			dirname  = "{0} {1}".format(self.i18n.MENU_SEASON_EPISODE, i)
+			iconpath = "{0}{1}.jpg".format(self.paths.DEFAULT_IMGDIR, i)
+			self.add_directory(dirname, str(i), PLUGIN_MODE_SEASON, iconpath)
+		xbmcplugin.endOfDirectory(self.phandle)
+
+	def create_featured(self):
+		jsonrsp = _http_get(self.helper.carousel())
+		promojson = _json.loads(jsonrsp)
+		for episode in promojson['results']:
+			self.add_episode(episode)
+
+	def create_random(self):
+		if not self.options.playrandom():
+			self.notify(self.i18n.WARNING_LOADING_RANDOM_EPISODE, WARNING_TIMEOUT_SHORT)
+		retries = 0
+		while retries < 10:
+			retries += 1
+			body = _http_get(self.helper.random_episode("random"))
+			rand = body.split("<link rel=\"canonical\" href=\"")[1].split("\" />")[0]
+			rand = rand.split(self.helper.random_episode("s"))[1].split("-")[0].split("e")
+			season  = int(rand[0])
+			episode = int(rand[1]) - 1
+
+			jsonrsp = _http_get(self.helper.season_data(season))
+			if self.options.audio(True) == "de":
+				# sp.de returns a JS instead of a JSON so i need to convert it
+				jsonrsp = jsonrsp.decode('utf-8')
+
+			seasonjson = _json.loads(jsonrsp)
+			episode_data = seasonjson['results'][episode]
+			if episode_data['_availability'] == "banned":
+				log_error("Found banned episode s{0}e{1}. trying again!".format(season, episode))
+				continue
+			if self.options.playrandom():
+				self.play_episode(episode_data['itemId'], episode_data['title'], episode_data['images'])
+			else:
+				self.add_episode(episode_data)
+			break
+
+	def create_search(self):
+		keyboard = xbmc.Keyboard('')
+		keyboard.doModal()
+		if (keyboard.isConfirmed()):
+			text = keyboard.getText().lower()
+			jsonrsp = _http_get(self.helper.search(text.replace(' ', '+')))
+			xbmcplugin.addSortMethod(self.phandle, xbmcplugin.SORT_METHOD_EPISODE)
+			seasonjson = _json.loads(jsonrsp)
+			for episode in seasonjson['results']:
+				self.add_episode(episode)
+
+	def create_episodes(self, season):
+		xbmcplugin.addSortMethod(self.phandle, xbmcplugin.SORT_METHOD_EPISODE)
+		jsonrsp = _http_get(self.helper.season_data(season))
+		if self.options.audio(True) == "de":
+			# sp.de returns a JS instead of a JSON so i need to convert it
+			jsonrsp = jsonrsp.decode('utf-8')
+		seasonjson = _json.loads(jsonrsp)
+		for episode in seasonjson['results']:
+			self.add_episode(episode)
+
+	def play_episode(self, url, title, thumbnail):
+		log_debug(url)
+		log_debug(title)
+		log_debug(thumbnail)
```

Comment:
reduce the log spamming

Reply:
i've commented the line 571 & 572, so they will never be called.